### PR TITLE
Add implementation and test to improve this mechanism.

### PR DESCRIPTION
**Pull request by Dwordcito**
## Description
This PR improve the interfaces, to accept a filepath to download a file in the API response.

For this change, the derived output mechanism is generalized in a file, this means that it is no longer specific for the get itself.

### CODE REVIEW

File: test/component/component_test.cpp [hunk 1]
```diff
@@ -113,7 +121,9 @@ TEST_F(ComponentTestInterface, DownloadFile)
 TEST_F(ComponentTestInterface, DownloadFileError)
 {
     auto callbackComplete = false;
-    HTTPRequest::instance().download(HttpURL("http://localhost:44441/invalid_file"), "./test.txt", [&](const std::string &result)
```
Comment (NahuFigueroa97) [inline test/component/component_test.cpp:116]: This is constantly repeated. You could put it in the setUp() of UrlRequestUnitTest. Correct it in all TC.

Comment (Dwordcito) [reply test/component/component_test.cpp:116]: I did not understand your comment, this line is part of this particular test. I don't see anything code common between the tests, that should be initialized.

Comment (Dwordcito) [reply test/component/component_test.cpp:116]: do you mean the initialization of callbackComplete? 

`auto callbackComplete = false;`

Comment (NahuFigueroa97) [reply test/component/component_test.cpp:116]: Exactly, sorry if I didn't explain myself well. It's just a detail.

**Review by jnasselle — APPROVED** [review]

**Pull request merged**